sim/firesim-lib/src/main/scala/bridges/TracerVBridge.scala

```diff
@@ -206,13 +206,24 @@ class TracerVBridgeModule(key: TracerVKey)(implicit p: Parameters)
       ),
     )
 
+    // performance optimization:
+    //
+    // if any of these cases is true, just spam out hPort.fromHost.hValid=1,
+    // because triggerDebit will never be set and (if trigger result is always
+    // 1) triggerCredit will be set the first cycle then always zero.
+    //
+    // false: trigger condition is unsatisfiable
+    val trigger_always_false = (triggerSelector === 1.U) && (triggerCycleCountStart > triggerCycleCountEnd)
+    // true: triggering is not enabled (i.e. selector = 0) OR using a cycle count trigger with start=0 and end=(Largest 64 Bit UInt)
+    val trigger_always_true = (triggerSelector === 0.U) || ((triggerSelector === 1.U) && (triggerCycleCountStart === 0.U) && (triggerCycleCountEnd === (BigInt("ffffffffffffffff", 16).U(64.W))))
+
     val tFireHelper = DecoupledHelper(streamEnq.ready, hPort.toHost.hValid, hPort.fromHost.hReady, initDone)
 
     val triggerReg = RegEnable(trigger, false.B, tFireHelper.fire())
     hPort.hBits.triggerDebit  := !trigger && triggerReg
     hPort.hBits.triggerCredit := trigger && !triggerReg
 
-    hPort.fromHost.hValid := tFireHelper.fire(hPort.fromHost.hReady)
+    hPort.fromHost.hValid := tFireHelper.fire(hPort.fromHost.hReady) || ((trigger_always_false || trigger_always_true) && initDone)
```

Comment:
I guess what i've been trying to say is that:
`tFireHelper.fire(hPort.fromHost.hReady` is likely introducing a combinational dependency on the value of a token coming in from `toHost`, so the "better" solution would be to put what essentially amounts to the `fromHost` token value of the credit and debit in a queue, and then drive hPort.fromHost.{hValid, hReady} with the head of that queue only. 

Finally you have to ensure that the queue is initialized with a single token (say the credits both set to false) after reset. 

This will be less brittle and ensure good FMR even when the trigger is being used.